src/btreemap/iter.rs

```diff
@@ -239,14 +325,15 @@ where
                 // If the key does not belong to the range, iteration stops.
                 if !self.range.contains(node.key(entry_idx)) {
                     // Clear all cursors to avoid needless work in subsequent calls.
-                    self.cursors = vec![];
+                    self.forward_cursors = vec![];
                     return None;
                 }
 
                 let res = map(&node, entry_idx);
+                self.range.0 = Bound::Excluded(node.key(entry_idx).clone());
```

Comment:
Updating the remaining range causes a small regression in the benchmarks. If I comment out this line everything stays a 0% change. But this is required in order to iterate backwards up to the correct bound.
I think in a real canister this would be reading keys from stable memory so the impact of setting this field on the heap will become negligible.

Reply:
I just mean that in the benchmarks the memory used is `VectorMemory` which is significantly faster than `Ic0StableMemory`, so the tiny regression in the benchmarks will become even smaller in reality when `Ic0StableMemory` is being used.

Reply:
I don't think this is accurate. The benchmarks are using canbench which is running real canisters. It wouldn't make much sense to do anything else as we wouldn't be benchmarking what would actually be running in production. Do you have evidence that a `VectorMemory` is used instead or was it an assumption of yours?

Reply:
Ah right my mistake then. This is my first time using `canbench` and when I look at the functions it doesn't appear as if they're running within real canisters but I guess there's some magic happening to make them do so.